DEPRECATIONS.md

```diff
@@ -0,0 +1,66 @@
+# Deprecations & Breaking Changes in KEDA
+
+This document describes how KEDA handles deprecations and breaking changes and should provide clarity to KEDA end-users on what to expect when upgrading to a new version of KEDA.
+
+If you are interested in the current open deprecations, we recommend checking our [GitHub Discussions](https://github.com/kedacore/keda/discussions/categories/deprecations) or the open [breaking changes](https://github.com/kedacore/keda/issues?q=is%3Aopen+label%3Abreaking-change+sort%3Aupdated-desc).
+
+## Our versioning strategy
+
+KEDA follows the deprecation strategy of Kubernetes and is allowed to make breaking changes in minor versions after a deprecation period.
+
+The KEDA runtime, which consists of the operator and metric server, will provide support on the deprecated features for 4 releases/1 year after the deprecation was announced. In the following release, the feature will be removed.
+
+Depending on the area of the change, KEDA might not wait 4 releases to introduce a breaking change because one of its dependencies such as Kubernetes or scaler dependencies. However, these are the only exceptions to the rule.
```

Comment:
This is a proposal by @zroubalik 
```suggestion
Depending on the area of the change, KEDA might not wait 2 releases to introduce a breaking change because one of its dependencies such as Kubernetes or scaler dependencies. However, these are the only exceptions to the rule.
```

Any preferences @jeffhollan @JorTurFer @ahmelsayed? This means supporting something for 6mo / 2 releases.

(however, we might want to explicitly say "6mo (equivalent of 2 releases)" in case we decide to start shipping faster)

Reply:
I can live with 4 (1 year) but I prefer 2 releases (6 months, so this is okay to me

Reply:
Personally, I'd always prefer longer deprecation cycles, but I don't want to deadlock the decision since I'm ok with either. 
though I'm a bit confused by the 6 month thing, seems Jorge you suggesting we guarantee a minimum of 6 months notice, while Tom you are saying 2 releases regardless of release cadence? or did I misunderstand that?

Reply:
I see, is that what you meant Tom? That sounds good to me

Reply:
Yes, that is correct. Updated in [`3dd6427` (#70)](https://github.com/kedacore/governance/pull/70/commits/3dd642773f590d22d5012de027e9d74ef86ce10f) which is purely based on time and uses releases as a reference